tpgtools/templates/resource.go.tmpl

```diff
@@ -302,7 +311,16 @@ func resource{{$.PathType}}Read(d *schema.ResourceData, meta interface{}) error
 	if err != nil {
 		return err
 	}
-	client := Create{{$.ProductType}}Client(config, userAgent, "")
+	{{- if $.HasProject }}
+	billingProject := project
+	{{- else }}
+	billingProject := ""
+	{{- end}}
+	// err == nil indicates that the billing_project value was found
+	if bp, err := getBillingProject(d, config); err == nil {
+		billingProject = bp
+	}
+	client := Create{{$.ProductType}}Client(config, userAgent, billingProject)
```

Comment:
not needed for this PR, just a general q: should we consider calling this `NewServiceClient` instead? That's more consistent with the interface for handwritten clients (`op, err := config.NewComputeBetaClient(userAgent).Instances.Insert(project, zone.Name, instance).Do()`) as well as fairly standard for constructor/factory methods in Go (although I'll admit, I can't find a specific style guide item to back that up)

Reply:
Ironically the reason I chose create was to specifically not conflict with those functions since they all share a namespace. Also, I like functions to begin with a verb since they do things? umm but yeah I suppose I could always do `New{{blah_blah}}Client` but I'd probably need to throw dcl in there to ensure no conflict. Let me know what you prefer!

Reply:
Yeah, let's do NewDCLServiceClient in that case! `NewX` matches other Go code much more closely.

Reply:
Sounds good! I'll change it in a subsequent pr